### PR TITLE
Add fake crafting card for AE

### PR DESCRIPTION
**Pull request by MCTBL**
https://github.com/user-attachments/assets/a1625364-6fc8-46aa-a6ab-42730abe62de

this upgrade can finish crafting job without receiving output

| item | fake crafting | normal |
| :-: | :-: | :-: |
| image | image | image |

### CODE REVIEW

**Comment by serenibyss** [comment]
Why does this have a button to toggle the mode? Shouldn't an interface having the card enable this functionality, then not having the card is like normal?

**Comment by miaowwwwww** [comment]
> Why does this have a button to toggle the mode? Shouldn't an interface having the card enable this functionality, then not having the card is like normal?

so that players have the option to not remove the card and to just toggle it off.

**Comment by MCTBL** [comment]
> Why does this have a button to toggle the mode? Shouldn't an interface having the card enable this functionality, then not having the card is like normal?

I'm not sure if there has any player would use this in some time, so i add it.

**Comment by serenibyss** [comment]
I don't see why we should add more to the UI when this card is already a toggle on its own

File: src/main/java/appeng/api/config/Settings.java [hunk 1]
```diff
@@ -83,7 +83,9 @@ public enum Settings {
 
     TERMINAL_FONT_SIZE(EnumSet.allOf(TerminalFontSize.class)),
 
-    INTERFACE_TERMINAL_SECTION_ORDER(EnumSet.allOf(StringOrder.class));
+    INTERFACE_TERMINAL_SECTION_ORDER(EnumSet.allOf(StringOrder.class)),
+
+    FAKE_CRAFTING_MOED(EnumSet.allOf(FakeCraftingMode.class));
```
Comment (lordIcocain) [inline src/main/java/appeng/api/config/Settings.java:88]: MOED

File: src/main/java/appeng/api/config/Upgrades.java [hunk 2]
```diff
@@ -43,7 +43,8 @@ public enum Upgrades {
     PATTERN_CAPACITY(1),
     ORE_FILTER(1),
     ADVANCED_BLOCKING(1),
-    LOCK_CRAFTING(1);
+    LOCK_CRAFTING(1),
+    FAKE_FRAFTING(1);
```
Comment (lordIcocain) [inline src/main/java/appeng/api/config/Upgrades.java:47]: FRAFTING

File: src/main/java/appeng/me/cluster/implementations/CraftingCPUCluster.java [hunk 3]
```diff
@@ -837,6 +854,15 @@ private void executeCrafting(final IEnergyGrid eg, final CraftingGridCache cc) {
                     }
 
                     if (this.remainingOperations == 0) {
+                        // if (m instanceof DualityInterface di && di.isFakeCraftingMode()) {
```
Comment (lordIcocain) [inline src/main/java/appeng/me/cluster/implementations/CraftingCPUCluster.java:857]: is this redundant?

File: src/main/java/appeng/api/config/Settings.java (hunk 1 above)
Comment (MCTBL) [reply src/main/java/appeng/api/config/Settings.java]: sry for my bad english

Comment (Dream-Master) [reply src/main/java/appeng/api/config/Settings.java]: Dont worry my English is not better :)

**Review by serenibyss — CHANGES_REQUESTED** [review]

**Comment by AdityaVG13** [comment]
> I wish to re-add the button, Cause tooltips can write description for the fake craft card. To tell the player how to use this card.

I have already added a quest for this as well

**Pull request merged**